packages/webapp/src/containers/Finances/saga.js

```diff
@@ -240,6 +246,82 @@ export function* tempEditExpenseSaga(action) {
   }
 }
 
+export const getRevenueTypes = createAction('getRevenueTypesSaga');
+
+export function* getRevenueTypesSaga() {
+  const { revenueTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+
+  try {
+    const result = yield call(axios.get, `${revenueTypeUrl}/farm/${farm_id}`, header);
+
+    yield put(getRevenueTypesSuccess(result.data));
+  } catch (e) {
+    console.log('failed to fetch revenue types from database');
+  }
+}
+
+export const deleteRevenueType = createAction('deleteRevenueTypeSaga');
+
+export function* deleteRevenueTypeSaga({ payload: id }) {
+  const { revenueTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+  try {
+    yield call(axios.delete, `${revenueTypeUrl}/${id}`, header);
+
+    yield put(deleteRevenueTypeSuccess(id));
+    yield put(enqueueSuccessSnackbar(i18n.t('message:REVENUE_TYPE.SUCCESS.DELETE')));
+  } catch (e) {
+    yield put(enqueueErrorSnackbar(i18n.t('message:REVENUE_TYPE.ERROR.DELETE')));
+  }
+}
+
+export const addCustomRevenueType = createAction('addRevenueTypeSaga');
+
+export function* addRevenueTypeSaga({ payload: { revenue_name } }) {
+  const { revenueTypeUrl } = apiConfig;
+  let { user_id, farm_id } = yield select(loginSelector);
+  const header = getHeader(user_id, farm_id);
+
+  const body = {
+    revenue_name,
+    farm_id: farm_id,
```

Comment:
I think this may no longer be needed and headers is used.

Reply:
I know we _just_ discussed this last week with Gursimran but I've already forgotten the details! As far as I can see it's going to still be needed in the POST for `hasFarmAccess` (right @gursimran-singh?) Or at least that's how it is currently still set up in the expense PR:

```
router.post(
  '/',
  hasFarmAccess({ body: 'farm_id' }),
  checkScope(['add:expense_types']),
  farmExpenseTypeController.addFarmExpenseType(),
);
```

Did I miss something from our discussion Friday?

Reply:
Oh yea I might have missed that piece.

Reply:
@kathyavini yes, `hasFarmAccess` middleware needs this to check farm access.